### DISCUSSION TITLE
HTML and XSLT

### DISCUSSION THREAD

**Discussion by jgburd79** [epeios-q37/atlas-python#6] [Q&A] [answered]
I have no knowledge of generating HTML from XSLT. What is the significance of using it and could this be expanded more in another example?

**Comment by epeios-q37** [comment, marked as answer]
In general, the larger the application, the smaller the size of the XML + XSL compared to the size of the HTML. Moreover, as the transformation of XML + XSL into HTML is done by the browser, the latter caches the XSL, so only XML is systematically transferred.
Also, since XSL is XML, you can use all the tools dedicated to XML with XSL (syntax highlighting in an editor, for example).
But, honestly, it's only possible to use XSL with the Atlas toolkit because I am used to work with XSL since a very long time. Depending on the complexity of the application, it's probably better for someone who doesn't already know XSL to generate the HTML by hand or with one of the many other existing template engines. Learning XSL can be quite difficult as it is a functional language.
The _Chatroom(s)_, _Notes_ and _TodoMVC_ applications use XSLT, and you will find plenty of examples of XSLT on the web, which can be used with the Atlas toolkit, as the latter provides tools to generate XML.
It would be interesting to compare XSLT with other template engines, but XSLT is the only template engine I never used...
